Replace System.out and System.err with logging

denizt opened the pull request:
These series of commits removes all `System.out` and `System.err` from the code replacing in with `java.util.logging`, except from `examples` and `tests`

There are also some atomic commits that removes dead/unused code.

denizt committed:
chore: remove unused debug method

denizt committed:
chore: sout to logging in SystemUtils

denizt committed:
chore: remove commented out code

denizt committed:
chore: remove commented out system.out from ActorSystem

denizt committed:
chore: add logging to Generator

denizt committed:
chore: replace system.out with logging in HealRegistry

denizt committed:
chore: replace system.out with logging in Scheduler

denizt committed:
chore: remove commented out system.out from MessageHolder

denizt committed:
chore: replace debug console output with logging in Stereotypes

denizt committed:
chore: log for real in Exceptions

denizt committed:
chore: add logging to ActorSystem exceptions

denizt committed:
chore: add logging to TcpReceiver exceptions

denizt committed:
chore: add logging to MulticastActorRegistry exceptions

This commit also removes an unused local variable

denizt committed:
chore: add logging to CacheAheadList exceptions

denizt committed:
chore: add logging to BaseActor exception

lucav76 commented on src/main/java/eu/lucaventuri/common/Exceptions.java:
```diff
@@ -147,19 +151,15 @@ public static void logShort(RunnableEx run) {
-            // log.error(t.getMessage(), t); Not working...
-            // FIXME: log for real
-            System.err.println(t);
+            logger.log(Level.FINEST, t.getMessage(), t);
```
Many of the statements on this class are wrong, and this is critical as this class is used a lot to decide how to print the exceptions:
- FINEST (except if the method is called silence or ignore) should probably be INFO: we want to see those message
- If a method is called xxxShort(), it means I do not want to see the stack trace, just the message and the type of the exception

lucav76 commented on src/main/java/eu/lucaventuri/common/SystemUtils.java:
```diff
@@ -226,8 +229,7 @@ public static long printTime(Runnable run, String description) {
         long time = System.currentTimeMillis() - start;
-
-        System.out.println(description + " : " + time + " ms");
+        logger.log(Level.FINE, description + " : " + time + " ms");
```
This method is called printTime(), I think it should always print and it is unnecessary to log.

lucav76 commented on src/main/java/eu/lucaventuri/common/SystemUtils.java:
```diff
@@ -253,14 +254,14 @@ public static long transferStream(InputStream is, OutputStream os, String echoLa
                 os.write(buffer, 0, read);
                 if (echoLabel != null && transferred < 128)
-                    System.out.println(echoLabel + ": " + new String(buffer, 0, (int) Math.min(read - transferred, 128)));
+                    logger.log(Level.FINE, echoLabel + ": " + new String(buffer, 0, (int) Math.min(read - transferred, 128)));
```
These are not intended as log, please leave the println

lucav76 commented on src/main/java/eu/lucaventuri/concurrent/AntiFreeze.java:
```diff
@@ -99,46 +99,14 @@ public static void main(String[] args) throws InterruptedException {
 
         try {
             System.out.println("Starting...");
-            /*AntiFreeze frz = new AntiFreeze(1500, 7000);
```
I might need this code if I need to test the anti freeze again, in different scenarios, so I am not sure I want to delete it.

lucav76 commented on src/main/java/eu/lucaventuri/fibry/ActorSystem.java:
```diff
@@ -174,7 +176,7 @@ public <T> WeightedActor<T, S> newWeightedPool(Consumer<T> actorLogic, int numPe
                     actorLogic.accept(msg.message);
                 } catch (Exception e) {
-                    System.err.println("Weighted actor: " + e);
+                    logger.log(Level.FINEST, "Weighted actor", e);
```
Why do you want to hide all these messages?
Please use INFO, not FINEST. In all the lines. Else it would be impossible to notice and debug any issue.

lucav76 commented on src/main/java/eu/lucaventuri/fibry/BaseActor.java:
```diff
@@ -312,7 +316,7 @@ public final void processMessages() {
                         Exceptions.logShort(() -> autoHealing.onInterruption.accept(e));
                     } else {
-                        System.err.println(e);
+                        logger.log(Level.FINEST, e.getMessage(), e);
```
Please, do not print the stack trace if the original code was not doing it. I do not want to pollute the log files unnecessary.

denizt committed:
fix: change level of logging for Exceptions

Methods suffixed with short no longer log the stack trace. We also
bump the level to INFO which is default, unless overriden by a
config file.

Silent methods now log the exceptions with the stack trace only
for FINEST level.

denizt committed:
fix: print time in printTime function in SystemUtils

Also add a FINEST level of logging for caught exceptions.

denizt committed:
chore: bring back commented code

Add it as an extracted method and comment out the call instead.

denizt committed:
chore: bump logging level to INFO for ActorSystem

denizt committed:
chore: don't log the stack trace in BaseActor
